arelle/Cntlr.py

```diff
@@ -291,6 +294,24 @@ def __init__(
         PackageManager.init(self, loadPackagesConfig=hasGui)
 
         self.startLogging(logFileName, logFileMode, logFileEncoding, logFormat)
+        self.errorManager = ErrorManager(self.modelManager, logging._checkLevel("INCONSISTENCY")) # type: ignore[attr-defined]
+
+    def error(self, codes: Any, msg: str, level: str = "ERROR", fileSource: FileSource | None = None, **args: Any) -> None:
+        if self.logger is None:
+            return
+        self.errorManager.log(
```

Comment:
Similar to addToLog, this might be called before the error manager is initialized, so need to guard against that. It also shouldn't discard the error if that happens.